sandbox/lua/modules/line_protocol.lua

```diff
@@ -0,0 +1,207 @@
+    module to generate a table of Carbon line protocol messages that are
+    derived from the dynamic fields in a Heka message. Base fields or
+    dynamic fields can be used in the metric name portion of the message.
+    Configuration is implemented in the encoders that utilize this module.
+    This function takes no arguments.
+
+**influxdb_line_msg()**
+    Wrapper function that calls others within this module and the field_util
+    module to generate a table of InfluxDB line protocol messages that are
+    derived from the base or dynamic fields in a Heka message. Base fields or
+    dynamic fields can be used in the metric name portion of the message and
+    included as tags if desired.  Configuration is implemented in the encoders
+    that utilize this module.  This function takes no arguments.
+
+--]=]
+
+local field_util = require "field_util"
+local math = require "math"
+local read_config = read_config
+local read_message = read_message
+local read_next_field = read_next_field
+local pairs = pairs
+local string = require "string"
+local table = require "table"
+local tostring = tostring
+local type = type
+
+local M = {}
+setfenv(1, M) -- Remove external access to contain everything in the module.
+
+if not timestamp_precision then
+    timestamp_precision = "ms"
+end
+if not decimal_precision then
+    decimal_precision = "6"
+end
+if not name_prefix then
+    name_prefix = ""
+end
+if not name_prefix_delimiter then
+    name_prefix_delimiter = ""
+end
+if not value_field_name then
+    value_field_name = "value"
+end
+
+-- Remove blacklisted fields from the set of base fields that we use, and
+-- create a table of dynamic fields to skip.
+local skip_fields_str = read_config("skip_fields") or ""
+local skip_fields, skip_fields_all_base = field_util.field_map(skip_fields_str)
+local used_base_fields = field_util.used_base_fields(skip_fields)
+
+-- Create and populate a table of fields to be used as tags
+local tag_fields_str = read_config("tag_fields") or "**all_base**"
+local used_tag_fields, tag_fields_all_base, tag_fields_all = field_util.field_map(tag_fields_str)
+
+local function tags_table()
+    -- Initialize and populate the table of tags to include in the
+    -- InfluxDB write API message
+    -- Convert value to a string as this is required by the API
+    local message_tags = {}
+    for field in pairs(field_util.base_fields_list) do
+        local base_field_value = read_message(field)
+        field = field:gsub("([ ,])", "\\%1")
+        if (tag_fields_all or tag_fields_all_base)
+            and field_util.base_fields_tag_list[field] and type(base_field_value) ~= "number" then
+            table.insert(message_tags, field.."="..tostring(base_field_value))
+        elseif used_tag_fields[field] and field_util.base_fields_tag_list[field]
+            and type(base_field_value) ~= "number" then
+            table.insert(message_tags, field.."="..tostring(base_field_value))
+        end
```

Comment:
I actually cleaned this up and moved the functionality into the `points_tags_tables` function, since it really only needs to loop through `base_fields_tag_list` (the whole point of that table even existing).  I kept the for loop inside a conditional that checks if we are working with `carbon_format` so it will never run for Carbon but with always run for InfluxDB if there is configuration in place to add tags.  It then only matches if any of the previously checked conditions is true.  There is a little code duplication, but net result is improved, IMO.
(updates to the PR will be made later after this comment is posted).